train_cnn_lstm_enhanced.py

```
@@ -0,0 +1,831 @@
+            logger.warning(f"Trial failed: {e}")
+            return float("inf")
+    
+    def optimize(self) -> Dict[str, Any]:
+        """Run hyperparameter optimization."""
+        
+        logger.info(f"🔍 Starting hyperparameter optimization with {self.n_trials} trials...")
+        
+        study = optuna.create_study(direction="minimize")
+        study.optimize(self.objective, n_trials=self.n_trials)
+        
+        logger.info(f"✅ Optimization completed!")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)